File: convert/example1_todo/java/todo/src/main/java/dev/practice/todo/domain/Todo.java
package dev.practice.todo.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Todo {

    /**
     * 할일 리스트 도메인 객체
     */

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Lob
    private String description;
    private Boolean done; // 할 일이 끝났는지
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @Builder
    private Todo(Long id, String title, String description, Boolean done, LocalDateTime createdAt, LocalDateTime updatedAt) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.done = done;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    public void update(String title, String description, Boolean done) {
        this.title = title;
        this.description = description;
        this.done = done;
        this.updatedAt = LocalDateTime.now();
    }
}

File: practice/convert/example1_todo/kotlin/todo/src/main/java/dev/practice/todo/domain/Todo.kt
package dev.practice.todo.domain

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Lob
import java.time.LocalDateTime

//https://multifrontgarden.tistory.com/272
@Entity
class Todo(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    var title: String,

    @Lob
    var description: String,
    var done: Boolean,
    var createdAt: LocalDateTime,
    var updatedAt: LocalDateTime? = null,
) {
    fun update(title: String, description: String, done: Boolean) {
        this.title = title
        this.description = description
        this.done = done
        this.updatedAt = LocalDateTime.now()
    }
}